pinot-common/src/test/java/org/apache/pinot/common/tier/TierSegmentSelectorTest.java

```diff
@@ -46,8 +45,8 @@ public void testTimeBasedSegmentSelector() {
     // offline segment
     String segmentName = "segment_0";
     String tableNameWithType = "myTable_OFFLINE";
-    OfflineSegmentZKMetadata offlineSegmentZKMetadata = new OfflineSegmentZKMetadata();
-    offlineSegmentZKMetadata.setSegmentName(segmentName);
+    SegmentZKMetadata offlineSegmentZKMetadata = new SegmentZKMetadata(segmentName);
+    offlineSegmentZKMetadata.setSegmentType(CommonConstants.Segment.SegmentType.OFFLINE);
```

Comment:
It is called within the constructor of `OfflineSegmentZKMetadata` and `RealtimeSegmentZKMetadata` which is being cleaned up in this PR. It is already deprecated and will be removed in the next release, so I didn't put it into the constructor so that it can be easily cleaned up